File: mpp-proiect-repository-cristinahendre/lab6(web)/Java/lab6/modul/DomainModule/src/main/java/Domain/Validators/DonatieValidator.java
package Domain.Validators;

import Domain.Donatie;

public class DonatieValidator implements  Validator<Donatie> {

    /**
     * Valideaza o donatie
     * @param entity -> donatia
     * @throws ValidationException
     */
    @Override
    public void validate(Donatie entity) throws ValidationException {
        if(entity.getSuma_donata() < 0)
            throw new ValidationException("Ati donat o suma negativa.");

        if(entity.getCaz() == null)
            throw new ValidationException("Introduceti un caz");

        if(entity.getDonator() == null)
            throw new ValidationException("Donatorul nu este introdus");
    }
}

File: mpp-proiect-repository-cristinahendre/C#/Cerinta1/Cerinta1/Domain/Validator/DonatieValidator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Cerinta1.Domain.Validator
{
    class DonatieValidator : IValidator<Donatie>
    {
        /*
          * Valideaza o donatie
         * @param entity -> donatia
         * @throws ValidationException
         */
        public void Validate(Donatie entity)
        {
            if (entity.Suma_donata < 0)
                throw new ValidationException("Ati donat o suma negativa.");
        }
    }
}
